cumulus_library/template_sql/shared_macros/syntax.sql.jinja

```diff
@@ -57,7 +60,11 @@ worry about the slightly more performant regex based matching we want to use
 in most cases.
 -#}
 {%- macro like(field, match_string) -%}
-REGEXP_LIKE({{ field }}, {{ wildcard_to_regex(match_string) }})
+REGEXP_LIKE({{ field }}, '^{{ wildcard_to_regex(match_string) }}$')
+{%- endmacro -%}
+
+{%- macro ilike(field, match_string) -%}
+REGEXP_LIKE({{ field }}, '.*(?i:{{ wildcard_to_regex(match_string) }}).*$')
```

Comment:
I don't like that `ilike` does TWO things different from `like` - the case insensitive thing makes sense and is obvious. It would _not_ be obvious to me that it also changes how the regex is anchored. Especially since the `ILIKE` SQL keyword doesn't work like that (at least in duckdb's implementation).

If that piece of this is important, can we instead add a new set of methods? `search` and `isearch` maybe? Whatever names make sense.

Reply:
Yeah, good call - I did that as a convenience for the vocab piece but it's probably not good behavior.